src/Core/Indexes/Octree.cs

```diff
@@ -70,17 +70,40 @@ protected new void CreateChildren()
         /// </summary>
         /// <param name="envelope">The maximum indexed region.</param>
         public Octree(Envelope envelope)
-            : base(envelope)
         {
+            this.root = new OctreeNode(envelope);
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="Octree" /> class.
         /// </summary>
         /// <param name="geometries">The geometries to add to the tree.</param>
         public Octree(IEnumerable<IBasicGeometry> geometries)
-            : base(geometries)
         {
+            Envelope bound = Envelope.FromEnvelopes(geometries.Select(geometry => geometry.Envelope));
+            this.root = new OctreeNode(bound);
+
+            this.Add(geometries);
+        }
+
+        /// <summary>
+        /// Clears all geometries from the index.
+        /// </summary>
+        public new void Clear()
```

Comment:
Since `Clear()` has public visibility, defining as new, instead of overriding can easily lead to errors, e.g.:

```csharp
QuadTree tree = new Octree();
tree.Clear(); // method in QuadTree is called
```